### ISSUE TITLE
warn when layer name doesn't match a datasource layer

### ISSUE THREAD

**Issue by meetar** [tangrams/tangram#640]
When a layer's `data` block doesn't specify a layername, and the layer's name doesn't match a datasource layer, log a warning to that effect.

Check whether there's any case when a datasource might not have a named layer, eg standalone untiled GeoJSON, and make a provision in that case.

**Comment by pnorman** [comment]
Isn't a MVT with no layers just an empty file? This sounds like it would warn on lots of valid situations.

**Comment by meetar** [comment]
I think the more common case would be datasources without layers, which are certainly valid. This as you point out would lead to lots of extraneous warnings.

So a better case for the new warning would be:
- the datasource has named layers
- a layer's `data` block doesn't specify a `layer`
- the layer's name doesn't match a datasource layer

Currently this just fails silently.

**Comment by pnorman** [comment]
Wouldn't this still be a common failure, e.g. if you are rendering a map with a POI layer and you're viewing somewhere with no POIs?

**Comment by bcamper** [comment]
If I understand the conversation, this is why I hadn't previously added a
warning like this. With tiled data in particular, you can't really know
where a given layer will or won't be present. I suppose a TileJSON could be
used to validate whether a layer is valid, but we haven't had a strong need
to support that yet.

What are the cases where the current behavior is a problem? Are you trying
to catch spelling errors in layer names?

**Comment by meetar** [comment]
> rendering a map with a POI layer and you're viewing somewhere with no POIs?
Of course you're right, for some reason when I opened the issue I was imagining that an empty tile would contain a set of empty layers, which is clearly incorrect.

Yes, the original problem case was mismatched or nonexistent layer names, but without some kind of schema to reference there's no way to tell whether a given tile ought to have a given layer.

Closing as unworkable – thanks for the comments!
